peps/pep-0752.rst

```diff
@@ -0,0 +1,399 @@
+  installer and tool has to be modified to account for scopes.
+* `NuGet <https://www.nuget.org>`__ has the concept of
+  `package ID prefix reservation`__ which was
+  `introduced`__ primarily to satisfy users wishing to know where a package
+  came from. A package name prefix may be reserved for use by one or more
+  owners. Every reserved package has a special indication
+  `on its page <https://www.nuget.org/packages/Google.Cloud.Storage.V1>`__ to
+  communicate this. After reservation, any upload with a reserved prefix will
+  fail if the user is not an owner of the prefix. Existing packages that have a
+  prefix that is owned may continue to release as usual. This approach has the
+  highest amount of backwards compatibility because only modifications to
+  indices like PyPI are required and installers do not need to change.
+
+__ https://blog.npmjs.org/post/116936804365/solving-npms-hard-problem-naming-packages
+__ https://docs.npmjs.com/package-scope-access-level-and-visibility
+__ https://learn.microsoft.com/en-us/nuget/nuget-org/id-prefix-reservation
+__ https://devblogs.microsoft.com/nuget/Package-identity-and-trust/
+
+This PEP specifies the NuGet approach of authorized reservation across a flat
+namespace for the following reasons:
+
+* Causing churn for the community is a hard blocker.
+* The NPM approach has the potential to cause confusion for users if we allow
+  unscoped names. Our community has chosen to normalize separator characters
+  and so ``@aws/s3`` would likely be confused with ``@aws-s3``.
+
+Granting Namespaces
+-------------------
+
+PyPI has been understaffed, receiving the first `dedicated specialist`__ in
+July 2024. Due to lack of resources, user support has been lacking for
+`package name claims <https://discuss.python.org/t/27436/19>`__,
+`organization requests <https://discuss.python.org/t/33764/15>`__,
+`storage limit increases <https://discuss.python.org/t/54035>`__,
+and even `account recovery <https://discuss.python.org/t/43422/122>`__.
+
+__ https://pyfound.blogspot.com/2024/07/announcing-our-new-pypi-support.html
+
+Reserving the ability to claim prefixes for
+`corporate organizations <corp-orgs_>`_ provides the following benefits:
+
+* PyPI would have a constant source of funding for support specialists,
+  infrastructure maintenance and new features.
+* Although each application would require independent review, less human
+  feedback would be required because the process to approve a paid organization
+  already bestows a certain amount of trust.
+
+Specification
+=============
+
+Organizations MAY reserve one or more namespaces. Such reservations neither
+confer ownership nor grant special privileges to existing packages.
+
+.. _naming:
+
+Naming
+------
+
+A namespace MUST be a `valid`__ project name and `normalized`__ internally e.g.
+``foo.bar`` would become ``foo-bar``. The user facing namespace (e.g. in UI
+tooltips) MUST preserve the original casing as defined during reservation.
```

Comment:
Ah, thank you for mentioning normalization.

Suggestion: "... MUST preserve the original, pre-normalization package name given in the package metadata."

Reply:
This refers to the namespace that would be requested on the application rather than embedded in an artifact. I did change `casing` to `pre-normalized text` though!